server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java

```diff
@@ -61,6 +67,44 @@
 import spark.Service;
 
 public class UserRoutes implements Routes {
+    interface UserCondition {
+        boolean satisfyCondition(UserResponse user);
+    }
+
+    static class HasNoMailboxesCondition implements UserCondition {
+        private final UserMailboxesService userMailboxesService;
+
+        public HasNoMailboxesCondition(UserMailboxesService userMailboxesService) {
+            this.userMailboxesService = userMailboxesService;
+        }
+
+        @Override
+        public boolean satisfyCondition(UserResponse user) {
+            try {
+                return userMailboxesService.listMailboxes(Username.of(user.getUsername())).isEmpty();
+            } catch (UsersRepositoryException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    static class HasNotAccessAllSystemMailboxesCondition implements UserCondition {
+        private final UserMailboxesService userMailboxesService;
+
+        public HasNotAccessAllSystemMailboxesCondition(UserMailboxesService userMailboxesService) {
+            this.userMailboxesService = userMailboxesService;
+        }
+
+        @Override
+        public boolean satisfyCondition(UserResponse user) {
+            try {
+                List<Mailbox> mailboxes = userMailboxesService.listMailboxes(Username.of(user.getUsername()));
+                return mailboxes.size() == 1 && DefaultMailboxes.INBOX.equals(mailboxes.getFirst().getName());
```

Comment:
No I wishes we verifies that for each system mailbox it exist onto the user account and return false if it is not the case. Nothing specific about INBOX.

Reply:
I checked the case "User never logged in but receives emails" by integration test and see that when user never login but still receive some emails, the "INBOX" mailbox is created automatically and this is the only mailbox created.

Therefore, I think this code is correct.

Reply:
Please understand that "user misses some system mailboxes" is not logically equivalent to "user only has an INBOX" !